### PR TITLE
HPCC-14184 Document the TRACE activty

### PR DESCRIPTION
**Pull request by JamesDeFabia**
Signed-off-by: Jim DeFabia <jamesdefabia@lexisnexis.com>

@g-pan Please review for formatting and include structure
@richardkchapman Please review for technical accuracy

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by g-pan** [comment]
Reviewed content and tests; approved.

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-TRACE.xml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        <colspec />
+
+        <tbody>
+          <row>
+            <entry><emphasis>attrname</emphasis></entry>
+
+            <entry>Optional. The action name, which turns the action into an
+            attribute definition, therefore not executed until the
+            <emphasis>attrname</emphasis> is used as an action.</entry>
```
Comment (richardkchapman) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-TRACE.xml]: This is wrong and misleading. TRACE is an expression, and an attrname just means it's a named expression. Using an expression without an attrname doesn't turn it into an action (actions are specific types of expression that do not return a result...)

File: docs/ECLLanguageReference/ECLR_mods/BltInFunc-TRACE.xml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+          <row>
+            <entry><emphasis>options</emphasis></entry>
+
+            <entry>Optional. One or more of the options listed below.</entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </informaltable></para>
+
+  <para>The <emphasis role="bold">TRACE </emphasis>action defines tracing to
```
Comment (richardkchapman) [inline docs/ECLLanguageReference/ECLR_mods/BltInFunc-TRACE.xml]: A paragraph on the wider purpose of TRACE would be nice - i.e. you can add TRACE to your code at interesting junctures without any impact on performance, then (should you need to investigate the behaviour) enable them without modifying your code just by setting an option or stored boolean.

**Comment by richardkchapman** [comment]
@JamesDeFabia Couple of comments

**Comment by JamesDeFabia** [comment]
@richardkchapman Edits made, back to you

**Pull request merged**